### PR TITLE
Better main memory hex view

### PR DESCRIPTION
**Pull request by maximilien-noal**
Implements a better, faster, more easily readable main memory hex view.

image

### CODE REVIEW

File: src/Spice86/MemoryWrappers/MemoryReadOnlyBitRangeUnion.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    public BitRangeUnion.Enumerator GetEnumerator() {
+        var bitRangeUnion = new BitRangeUnion();
+        // Multiply by 8 to convert from bytes to bits
+        bitRangeUnion.Add(new BitRange(0, _memory.Length * 8));
+        return bitRangeUnion.GetEnumerator();
+    }
+
+    public bool IntersectsWith(BitRange range) {
+        var enclosingRange = new BitRange(0, _memory.Length * 8);
```
Comment (Kender2) [inline src/Spice86/MemoryWrappers/MemoryReadOnlyBitRangeUnion.cs:36]: Why create a new BitRange and not use the existing EnclosingRange property?

Comment (JorisVanEijden) [inline src/Spice86/MemoryWrappers/MemoryReadOnlyBitRangeUnion.cs:36]: Why create a new BitRange and not use the existing EnclosingRange property?

File: src/Spice86/UserControls/AddressColumn.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+
+    /// <inheritdoc />
+    public override TextLine? CreateTextLine(VisualBytesLine line) {
+        if (HexView is null)
+            throw new InvalidOperationException();
+
+        ulong address = line.Range.Start.ByteIndex;
+        ushort segment = ConvertUtils.ToAbsoluteSegment((uint)address);
```
Comment (JorisVanEijden) [inline src/Spice86/UserControls/AddressColumn.cs:56]: Is it useful to even display segment:offset if we can't use the same values that the game does?

File: src/Spice86/UserControls/AddressColumn.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    private TextLine? CreateTextLine(string text) {
+        if (HexView is null)
+            return null;
+
+        GenericTextRunProperties properties = GetTextRunProperties();
+        return TextFormatter.Current.FormatLine(
+            new SimpleTextSource(text, properties),
+            0,
+            double.MaxValue,
```
Comment (JorisVanEijden) [inline src/Spice86/UserControls/AddressColumn.cs:73]: can you add parameter names to these numbers to make reviewing on the web easier?

File: src/Spice86/Views/DebugWindow.axaml [hunk 4]
```diff
@@ -1,519 +1,310 @@
...
+                                <DataGridTextColumn Binding="{Binding Bytes}" Header="Bytes" />
+                                <DataGridTextColumn Binding="{Binding Length}" Header="Length" />
+                                <DataGridTextColumn Binding="{Binding FlowControl}" Header="FlowControl" />
                             </DataGrid.Columns>
                             <TextElement.FontFamily>
-                                <OnPlatform Default="{StaticResource RobotoMonoFont}">
-                                    <On Options="Windows" Content="{StaticResource ConsolasFont}" />
-                                </OnPlatform>
+                                <OnPlatform Default="{StaticResource RobotoMonoFont}" />
                             </TextElement.FontFamily>
```
Comment (JorisVanEijden) [inline src/Spice86/Views/DebugWindow.axaml:128]: Isn;t the TextElement.FontFamily already set to this right before the DataGrid?

File: src/Spice86/MemoryWrappers/EmulatedMemoryStream.cs
Comment (JorisVanEijden) [inline src/Spice86/MemoryWrappers/EmulatedMemoryStream.cs:1]: The file was moved, but the namespace was not adjusted.

File: src/Spice86/UserControls/AddressColumn.cs (hunk 2 above)
Comment (maximilien-noal) [reply src/Spice86/UserControls/AddressColumn.cs:56]: Not really, no. Remove it ?

File: src/Spice86/MemoryWrappers/MemoryReadOnlyBitRangeUnion.cs (hunk 1 above)
Comment (maximilien-noal) [reply src/Spice86/MemoryWrappers/MemoryReadOnlyBitRangeUnion.cs]: Fixed.

File: src/Spice86/UserControls/AddressColumn.cs (hunk 2 above)
Comment (maximilien-noal) [reply src/Spice86/UserControls/AddressColumn.cs]: Removed. I use the OffsetColumn for the nuget package.

File: src/Spice86/UserControls/AddressColumn.cs (hunk 3 above)
Comment (maximilien-noal) [reply src/Spice86/UserControls/AddressColumn.cs]: File removed.

File: src/Spice86/MemoryWrappers/EmulatedMemoryStream.cs
Comment (maximilien-noal) [reply src/Spice86/MemoryWrappers/EmulatedMemoryStream.cs:1]: Fixed.

File: src/Spice86/Views/DebugWindow.axaml (hunk 4 above)
Comment (maximilien-noal) [reply src/Spice86/Views/DebugWindow.axaml]: Fixed.

**Pull request merged**